src/tests/async/valuetask-source/valuetask-source-stub.cs

```diff
@@ -0,0 +1,58 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Threading.Tasks.Sources;
+using Xunit;
+
+public class ValueTaskSourceAndStubs
+{
+    [Fact]
+    public static void EntryPoint()
+    {
+        SynchronizationContext.SetSynchronizationContext(new MySyncContext());
```

Comment:
I think we should set this inside `TestAsync` so that it doesn't leak out into the test infrastructure and other tests.